### PR TITLE
Update debugging docs a bit

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: dev/doc/debugging.md [hunk 1]
```diff
@@ -22,65 +22,12 @@ Debugging from Coq toplevel using Caml trace mechanism
...
-   - `backtrace` or `bt` to see the call stack
-   - `step` or `s` goes into called functions; `next` or `n` skips over them
-   - `list` or `li` shows the code just before and after the current stack frame
-   - `print <var>` or `p <var>` to see the value of a variable
-Note that `make byte` doesn't recompile .v files.  `make` recompiles all of them if there
-are changes in any .ml file--safer but much slower.
+Debugging with ocamldebug from Emacs or command line
+====================================================
+
+See build-system.dune.md#ocamldebug
```
Comment (Zimmi48) [inline dev/doc/debugging.md:28]:
```suggestion
See [build-system.dune.md#ocamldebug](build-system.dune.md#ocamldebug)
```

File: dev/doc/debugging.md [hunk 2]
```diff
@@ -22,65 +22,12 @@ Debugging from Coq toplevel using Caml trace mechanism
...
+
+Remember that you should not use configure unless calling dune through the legacy build system.
```
Comment (ejgallego) [inline dev/doc/debugging.md:30]: This is too strong, running configure with dune is supported and required in some cases; IMHO the correct wording is "you don't need to run configure when ..."

File: Makefile.dev [hunk 3]
```diff
@@ -37,46 +26,39 @@ printers: $(CORECMA) $(DEBUGPRINTERS)
...
-structures: $(filter theories/Structures/%, $(THEORIESVO))
-vectors: $(filter theories/Vectors/%, $(THEORIESVO))
-msets: $(filter theories/MSets/%, $(THEORIESVO))
-compat: $(filter theories/Compat/%, $(THEORIESVO))
+theories:=$(VO_OUT_DIR)theories
+
+init: $(filter $(theories)/Init/%, $(THEORIESVO))
+logic: $(filter $(theories)/Logic/%, $(THEORIESVO))
+arith: $(filter $(theories)/Arith/%, $(THEORIESVO))
+bool: $(filter $(theories)/Bool/%, $(THEORIESVO))
```
Comment (jfehrle) [inline Makefile.dev:40]: If the order doesn't matter, might be nice to list these in alphabetical order.

File: Makefile.make [hunk 4]
```diff
@@ -122,7 +122,7 @@ submake:
 	$(MAKE) $(MAKE_OPTS) -f Makefile.build $(MAKECMDGOALS)
 
 noconfig:
-	@echo "Please run ./configure first" >&2; exit 1
+	@echo "Please run ./configure first or use Makefile.dune" >&2; exit 1
```
Comment (herbelin) [inline Makefile.make:125]: I would suspect that it is worth being more explicit.
```suggestion
	@echo 'Please run ./configure first to continue building with make' >&2
	@echo 'or pass option "-f Makefile.dune" to let dune manage the build' >&2; exit 1
```

File: Makefile.dev (hunk 3 above)
Comment (SkySkimmer) [reply Makefile.dev:40]: The order is vaguely in dependency order, at least with init first.

Comment (SkySkimmer) [reply Makefile.dev]: resorted regardless

File: dev/doc/debugging.md [hunk 5]
```diff
@@ -22,65 +22,10 @@ Debugging from Coq toplevel using Caml trace mechanism
...
-8. See the ocamldebug manual for more information.  A few points:
-   - use `break @ Printer 501` to set a breakpoint on line 501 in the Printer module (printer.ml).
-     `break` can be abbreviated as `b`.
-   - `backtrace` or `bt` to see the call stack
-   - `step` or `s` goes into called functions; `next` or `n` skips over them
-   - `list` or `li` shows the code just before and after the current stack frame
-   - `print <var>` or `p <var>` to see the value of a variable
-Note that `make byte` doesn't recompile .v files.  `make` recompiles all of them if there
-are changes in any .ml file--safer but much slower.
+See [build-system.dune.md#ocamldebug](build-system.dune.md#ocamldebug)
```
Comment (Alizter) [inline dev/doc/debugging.md:28]: Maybe update build-system.dune.md too? There is some information loss here.

Comment (ejgallego) [reply dev/doc/debugging.md:28]: Indeed, @SkySkimmer , could we keep this information somewhere?

Other than that, the PR is good to go IMHO.

Comment (SkySkimmer) [reply dev/doc/debugging.md:28]: What info?

Comment (ejgallego) [reply dev/doc/debugging.md:28]: The small "ocamldebug" tutorial.

Comment (SkySkimmer) [reply dev/doc/debugging.md:28]: I added a link to https://ocaml.org/learn/tutorials/debug.html#The-OCaml-debugger

File: dev/doc/debugging.md [hunk 6]
```diff
@@ -22,65 +22,10 @@ Debugging from Coq toplevel using Caml trace mechanism
...
-         run
-	 step
-         back
-         start
-	 next
-	 last
-	 print x (abbreviated into p x)
-	 ...
-   7. some hints: 
-
```
Comment (ejgallego) [inline dev/doc/debugging.md:48]: @SkySkimmer , these hints, even if likely outdated, are what I mean; IMHO they are nice, as they point to specific places in Coq's codebase.

Comment (SkySkimmer) [reply dev/doc/debugging.md:48]: updated

Comment (ejgallego) [reply dev/doc/debugging.md:48]: Thanks!

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**